### DISCUSSION TITLE
Calculate displacement

### DISCUSSION THREAD

**Discussion by Bach-Nguyen-Dinh** [jrowberg/i2cdevlib#745] [Q&A]
I am using example MPU6050_DMP6_using_DMP_V6.12. As I understand, after subtracting raw acceleration reading to gravity and transforming to reference frame coordinates, I can double integrate the acceleration to get the displacement in real world. Can anyone verify my method?

**Comment by jrowberg** [comment]
The theory is correct, but the practical error is enormous. Double-integrating a value from a $1 consumer IMU will inevitably result in lots of compounding inaccuracy over a very short period of time. Unless your use case is not very strict, or you have an uncommonly tightly controlled environment, this will probably not work very well.

**Reply by Bach-Nguyen-Dinh** [reply]
I just want to test the theory but i have a problem. At start i go straight from x direction then rotate about z axis 90 degree to y direction and go straight again. It measured fine at first going straight, but when the rotation happened, both x and y displacement changed a lots (it is supposed to just have some minor changes) and after that the reading goes all over the places (sometimes it is not event increase in y direction). Is this problem due to the quality of sensor or the calculation of coordinates transformation?
